common/src/main/java/uk/co/harieo/forceop/common/DataConverter.java

```diff
@@ -0,0 +1,70 @@
+package uk.co.harieo.forceop.common;
+
+public class DataConverter {
+
+	private static final char[] HEX_ARRAY = "0123456789ABCDEF".toCharArray();
+
+	/**
+	 * Converts an array of bytes into hex in string format
+	 *
+	 * @param bytes to convert to hex
+	 * @return the hex encoded string
+	 */
+	public static String convertToHex(byte[] bytes) {
+		char[] hexChars = new char[bytes.length * 2];
+		for (int j = 0; j < bytes.length; j++) {
+			int v = bytes[j] & 0xFF;
+			hexChars[j * 2] = HEX_ARRAY[v >>> 4];
+			hexChars[j * 2 + 1] = HEX_ARRAY[v & 0x0F];
+		}
+		return new String(hexChars);
+	}
+
+	/**
+	 * Converts an array of hexadecimal chars into their binary equivalent
+	 *
+	 * @param hexChars an event number of hexadecimal chars
+	 * @return a byte array representing the binary of the hexadecimal parameter
+	 */
+	public static byte[] convertHexToBinary(char[] hexChars) {
+		byte[] bytes = new byte[hexChars.length / 2];
+
+		int byteIndex = 0;
+		for (int i = 0; i < hexChars.length; i+=2) {
+			char firstChar = hexChars[i];
+			char secondChar = hexChars[i++];
```

Comment:
this won’t work. `i++` evaluates as the previous value. you want `++i` if you want that behavior

also, you don’t want to modify `i`, that’s a recipe for disaster since it’s your looping variable.

also, i’d recommend just using `i + 1`

Reply:
Weirdly, this worked when I tested it even though it shouldn't have